hazelcast/src/main/java/com/hazelcast/config/RingbufferStoreConfig.java

```diff
@@ -57,6 +58,9 @@ public RingbufferStore getStoreImplementation() {
     }
 
     public RingbufferStoreConfig setStoreImplementation(RingbufferStore storeImplementation) {
+        if (storeImplementation != null) {
+            setClassName(null);
```

Comment:
It seems that the opposite action is not performed: `setClassName` should set `storeImplementation` to `null`.